akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/internal/ReplicationImpl.scala

```diff
@@ -86,8 +86,10 @@ private[akka] object ReplicationImpl {
     val onlyLocalOriginTransformer = Transformation.empty.registerAsyncEnvelopeOrElseMapper(envelope =>
       envelope.eventMetadata match {
         case Some(meta: ReplicatedEventMetadata) =>
-          if (meta.originReplica == settings.selfReplicaId) Future.successful(envelope.eventOption)
-          else filteredEvent // Optimization: was replicated to this DC, don't pass the payload across the wire
+          if (settings.indirectReplication || meta.originReplica == settings.selfReplicaId)
+            Future.successful(envelope.eventOption)
+          else
+            filteredEvent // Optimization: was replicated to this replica, don't pass the payload across the wire
```

Comment:
This is the reason why I haven't enabled it by default. Would have a substantial performance overhead. We'll see where it is going when we support star topologies and such.

Reply:
It would actually be nice to configure this per replica, but here on the producer side we don't know which replica is the consumer (at least I couldn't see that)

Reply:
yes, need to think this through again. we want to support a full mesh in cloud where indirect-replication=off, and edge connecting in to one or a few cloud replicas. edge consumers need indirect-replication=on

Maybe we could look at the allReplicas (or otherReplicas) and use that for deciding indirect. If it's included in the defined allReplicas then off, otherwise on. However, again, here on the producer side we don't know the replicaId of the consumer. Maybe we have to add that?

Reply:
We want to indirectly replicate from cloud to edge, and from edge to other cloud replicas, so maybe it is more about knowing which replication goes/comes from where when deciding (I haven't completely thought this all the way through)?